crates/astria-composer/src/searcher/geth_collector.rs

```diff
@@ -1,3 +1,14 @@
+//! `GethCollector` is responsible for fetching pending transactions submitted to rollup
+//! nodes and then passing them downstream for the searcher to process.
+//!
+//! It is responsible for fetching pending transactions submitted to the rollup Geth nodes and then
+//! passing them downstream for the executor to process. Thus, a searcher can have multiple
+//! collectors running at the same time funneling data from multiple rollup nodes.
+//!
+//! `GethCollector` uses the <https://geth.ethereum.org/docs/interacting-with-geth/rpc/pubsub#newpendingtransactions>
```

Comment:
```suggestion
//! ## Note
//! This collector is likely specific to go-ethereum and only checked to work wit. It makes use of the [`eth_subscribe`](https://geth.ethereum.org/docs/interacting-with-geth/rpc/pubsub#newpendingtransactions) JSON-RPC with arguments shown below. It appears as if go-ethereum is the only ethereum node that documents this.
//! ```json
//! { "id": 1, "jsonrpc": "2.0", "method": "eth_subscribe", "params": ["newPendingTransactions"] }
//! ```
```

Reply:
thanks! this gave me a better sense on writing these docs